actionpack/lib/action_dispatch/routing/mapper.rb

```diff
@@ -1682,13 +1680,8 @@ def draw(name)
         #
         #     match 'path', to: 'controller#action', via: :post
         #     match 'path', 'otherpath', on: :member, via: :get
-        def match(path, *rest, &block)
-          if rest.empty? && Hash === path
-            options  = path
-            path, to = options.find { |name, _value| name.is_a?(String) }
-
-            raise ArgumentError, "Route path not specified" if path.nil?
-
+        def match(*path_or_actions, as: DEFAULT, via: nil, to: nil, controller: nil, action: nil, on: nil, defaults: nil, constraints: nil, anchor: nil, format: nil, path: nil, internal: nil, **mapping, &block)
+          if path_or_actions.count > 1
```

Comment:
I don't think the conditional here matches what the previous code was doing and the deprecation inside doesn't seem to match having multiple actions specified (which I'm also not clear on whether that should even be deprecated).